packages/flutter_tools/lib/src/dart/package_map.dart

```diff
@@ -34,7 +34,13 @@ File? findPackageConfigFile(Directory dir) {
       return candidatePackageConfigFile;
     }
     final Directory parentDir = candidateDir.parent;
-    if (fileSystem.identicalSync(parentDir.path, candidateDir.path)) {
+    // identicalSync requires that both paths exist; if we assume that the
+    // parentDir exists if the candidateDir exists, then we can just check that
+    // the candidate dir exists (if it doesn't, then identicalSync would throw
+    // which is probably not what findPackageConfigFile is intende to do).
+    //
+    // See https://github.com/flutter/flutter/issues/163901.
+    if (candidateDir.existsSync() && fileSystem.identicalSync(parentDir.path, candidateDir.path)) {
```

Comment:
This is just to check if we hit the root directory.

We should probably not need any filesystem access at all.

I think it would be better to do:
```suggestion
    if (fileSystem.path.equals(parentDir.path, candidateDir.path)) {
```

Which just textually compares the paths.

Reply:
Thanks! I wasn't sure if this was load bearing, but it shouldn't be.

As an added bonus I no longer need a comment :)